packages/sdk-metrics/src/export/PeriodicExportingMetricReader.ts

```diff
@@ -127,25 +126,20 @@ export class PeriodicExportingMetricReader extends MetricReader {
       );
     }
 
-    const doExport = async () => {
-      const result = await internal._export(this._exporter, resourceMetrics);
-      if (result.code !== ExportResultCode.SUCCESS) {
-        throw new Error(
-          `PeriodicExportingMetricReader: metrics export failed (error ${result.error})`
-        );
+    if (resourceMetrics.resource.asyncAttributesPending) {
+      try {
+        await resourceMetrics.resource.waitForAsyncAttributes?.();
+      } catch (e) {
+        api.diag.debug('Error while resolving async portion of resource: ', e);
+        globalErrorHandler(e);
```

Comment:
TIL about globalErrorHandler.

Reply:
globalErrorHandler is a poorly understood and (in my opinion) poorly thought out spec. I'd probably get rid of it if i had the chance.